akd/src/directory.rs

```diff
@@ -316,9 +365,23 @@ impl<S: Storage + Sync + Send> Directory<S> {
             name_hash_bytes,
             H::merge_with_int(H::hash(stale_bytes), version),
         ]);
-        let label_slice = hashed_label.as_bytes();
-        let hashed_label_bytes = convert_byte_slice_to_array(&label_slice);
-        NodeLabel::new(u64::from_ne_bytes(hashed_label_bytes), 64u32)
+        // let label_slice = hashed_label.as_bytes();
+        let message_vec = from_digest::<H>(hashed_label).unwrap();
+        let message: &[u8] = message_vec.as_slice();
+
+        // VRF proof
+        vrf.prove(&secret_key, message).unwrap()
+    }
+
+    /// Use this function to retrieve the VRF public key for this AKD.
+    pub fn get_public_key(&self) -> Vec<u8> {
+        // Initialization of VRF context by providing a curve
+        let mut vrf = ECVRF::from_suite(CipherSuite::SECP256K1_SHA256_TAI).unwrap();
+        // Inputs: Secret Key, Public Key (derived) & Message
+        let secret_key =
+            hex::decode("c9afa9d845ba75166b5c215767b1d6934e50c3db36e89b127b8a622b120f6721")
+                .unwrap();
+        vrf.derive_public_key(&secret_key).unwrap()
     }
 
     // FIXME: Make a real commitment here, alongwith a blinding factor.
```

Comment:
Probably add a ref here to the issue you opened related to commitments? So we can link it back